CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc

```diff
@@ -1200,6 +1201,33 @@ static protected function string GetAcademySlotBonusDisplayString (XComGameState
 	return class'X2StrategyElement_DefaultStaffSlots'.static.GetBonusDisplayString(SlotState, "%SKILL", Contribution);
 }
 
+static function PatchCovertActionPromotionRewards()
+{
+	local X2StrategyElementTemplateManager TemplateManager;
+	local X2DataTemplate DataTemplate;
+	local X2CovertActionTemplate ActionTemplate;
+	local int index;
+	
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+
+	foreach TemplateManager.IterateTemplates(DataTemplate)
+	{
+		ActionTemplate = X2CovertActionTemplate(DataTemplate);
+
+		if (ActionTemplate != none)
+		{
+			// Must be a for loop because a foreach loop is passed a copy of the array
+			for (index = 0; index < ActionTemplate.Slots.Length; index++)
+			{
+				if (default.arrAllowPromotionReward.Find(ActionTemplate.DataName) == INDEX_NONE)
```

Comment:
Move this check outside of the `for` loop, as there is no reason to check it on every iteration